core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1946,6 +1946,16 @@ package object config {
       .booleanConf
       .createWithDefault(false)
 
+  private[spark] val EXECUTOR_KILL_ON_FATAL_ERROR_DEPTH =
+    ConfigBuilder("spark.executor.killOnFatalError.depth")
+      .doc("The max depth of the exception chain in a failed task Spark will search for a fatal " +
+        "error to check whether it should kill an executor. 0 means not checking any fatal " +
+        "error, 1 means checking only the exception but not the cause, and so on.")
```

Comment:
Could you add Spark version, @zsxwing ? Since SPARK-33587 is filed as `Improvement`, `3.1.0`?
```scala
.version("3.1.0")
```

Reply:
done

Reply:
Thanks!